application/include/mmul.hpp

```diff
@@ -17,17 +17,17 @@
 using namespace Eigen;
 
 template <typename Executor,
-          typename execution::executor_of_type<inline_executor, Executor> = 0>
+          typename execution::instance_of_base<inline_executor, Executor> = 0>
 void mmul(Executor ex, MatrixXd &a, MatrixXd &b, MatrixXd &c) {
```

Comment:
Two of these should be parameters should be const.

Reply:
Sorry for asking this again, what were the argument for not using tag dispatching based overloads? This feels like a solid use case for it.

Reply:
Tag dispatching overload will only allow executors with a fixed set of properties and type to be passed.
Eg: 
```cpp
void mmul(inline_executor<oneway_t, single_t, blocking_t::always_t> &ex, MatrixXd &a, MatrixXd &b, MatrixXd &c){...}
```
The above function will not accept an inline_executor having a custom allocator.

Tag dispatching will certainly be useful in some cases, let me come back to this point on where it can be useful.

Reply:
If you want the flexibility, you can always template the method for the specific properties which you don't care about the type, in your example case you claimed you don't care about the allocator property of the executor

```cpp
template<typename allocator_t>
void mmul(inline_executor<oneway_t, single_t, blocking_t::always_t, allocator_t> &ex, MatrixXd &a, MatrixXd &b, MatrixXd &c){...}
```